controller_interface/src/chainable_controller_interface.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "controller_interface/chainable_controller_interface.hpp"
+
+#include <vector>
+
+#include "hardware_interface/types/lifecycle_state_names.hpp"
+#include "lifecycle_msgs/msg/state.hpp"
+
+namespace controller_interface
+{
+ChainableControllerInterface::ChainableControllerInterface() : ControllerInterface() {}
+
+bool ChainableControllerInterface::is_chainable() const { return true; }
+
+std::vector<hardware_interface::CommandInterface>
+ChainableControllerInterface::export_reference_interfaces()
+{
+  // TODO(destogl): force here the names for reference interface to begin with controller's names
+  // TODO(destogl): check the size of the reference interfaces storage and [reserve] it if needed
+  return on_export_reference_interfaces();
+}
+
+bool ChainableControllerInterface::set_chained_mode(bool chained_mode)
+{
+  bool result = false;
+
+  if (get_state().id() != lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE)
+  {
+    result = on_set_chained_mode(chained_mode);
+
+    if (result)
+    {
+      in_chained_mode_ = chained_mode;
+    }
+  }
+  else
+  {
+    RCLCPP_ERROR(
+      get_node()->get_logger(),
+      "Can not change controller's chained mode because it is no in '%s' state. "
```

Comment:
no -> now ?

Reply:
Thanks! It should be “not”.

Regarding other question, there is no sorting of controller right now. But if you load them in proper order, the approach will work. Of course, this is not something we can leave like that, and we have do some sorting.

Reply:
Thanks for the answer. Then to move forward, and maybe some proper warning, I believe that we should move forward to integrate it.